diff -u memcache-3.0.5.orig/memcache_ascii_protocol.c memcache-3.0.5/memcache_ascii_protocol.c
--- memcache-3.0.5.orig/memcache_ascii_protocol.c	2010-10-04 05:41:21.000000000 +0800
+++ memcache-3.0.5/memcache_ascii_protocol.c	2011-11-30 19:15:16.000000000 +0800
@@ -379,7 +379,7 @@
 	request->parse = mmc_request_parse_response;
 	
 	if (slabid) {
-		cmd_len = spprintf(&cmd, 0, "stats %s %ld %ld\r\n", type, slabid, limit);
+		cmd_len = spprintf(&cmd, 0, "stats %s %ld %ld 0\r\n", type, slabid, limit);
 	}
 	else if (type) {
 		cmd_len = spprintf(&cmd, 0, "stats %s\r\n", type);
